pkg/kubectl/cmd/util/clientcache.go

```diff
@@ -72,7 +74,28 @@ func (c *ClientCache) ClientConfigForVersion(version *unversioned.GroupVersion)
 		preferredGV = &versionCopy
 	}
 
-	negotiatedVersion, err := client.NegotiateVersion(c.defaultClient, &config, preferredGV, registered.RegisteredGroupVersions)
+	gvClient, err := client.New(&config)
+	if err != nil {
+		return nil, err
+	}
+	clientVersions := []unversioned.GroupVersion{}
+	for ix := range registered.RegisteredGroupVersions {
+		clientVersions = append(clientVersions, registered.RegisteredGroupVersions[ix])
+	}
+	// Add in any third party group/version resources
+	list, err := gvClient.ThirdPartyResources(api.NamespaceDefault).List(api.ListOptions{})
```

Comment:
forbidden's from this API request should be handled gracefully. I don't think I'd expect the whole method to fail.